wsapi/wsapiV2.go

```diff
@@ -722,40 +720,55 @@ func HandleV2RawData(state interfaces.IState, params interface{}) (interface{},
 		return nil, NewInvalidHashError()
 	}
 
+	dbase := state.GetDB()
 	var block interfaces.BinaryMarshallable
-	var b []byte
-
-	if block, _ = state.FetchECTransactionByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	} else if block, _ = state.FetchFactoidTransactionByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	} else if block, _ = state.FetchEntryByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	}
-
-	if b == nil {
-		dbase := state.GetDB()
-
-		// try to find the block data in db and return the first one found
+	switch hashkey.Type {
+	case HashTypeFCTTx:
+		block, _ = state.FetchFactoidTransactionByHash(h)
+	case HashTypeECTx:
+		block, _ = state.FetchECTransactionByHash(h)
+	case HashTypeEntry:
+		block, _ = dbase.FetchEntry(h)
+	case HashTypeEBlock:
+		block, _ = dbase.FetchEBlock(h)
+	case HashTypeECBlock:
+		block, _ = dbase.FetchECBlock(h)
+	case HashTypeFBlock:
+		block, _ = dbase.FetchFBlock(h)
+	case HashTypeABlock:
+		block, _ = dbase.FetchABlock(h)
+	case HashTypeDBlock:
+		block, _ = dbase.FetchDBlock(h)
+	default:
+		if block, _ = state.FetchFactoidTransactionByHash(h); block != nil {
+			break
+		}
+		if block, _ = state.FetchECTransactionByHash(h); block != nil {
+			break
+		}
+		if block, _ = dbase.FetchEntry(h); block != nil {
+			break
+		}
```

Comment:
You mentioned that query for Entry is probably the most common so you could optimize the order by putting Entry first. Which I agree. Why did you still put tx queries first?

Reply:
When I mentioned that I forgot about tx hashes. I'm not sure which should take precedence: tx hashes or entry hashes. Perhaps entries should come before txs. But I think txs should come before the rest for some gut feeling reason??

Reply:
Ok. I guess it depends on the kind of application, if it's one that work with the monetary side of Factom (like an exchange) or with the data side (pretty much all other uses cases). No big deal but I think I'd still put Entry first.

Reply:
The amount of people using get-raw in the first place is pretty low, so if they are looking for an optimization they can just specify the hash type.